### PR TITLE
Merge release branch into master

### PR DESCRIPTION
**Pull request by gunan**

### CODE REVIEW

**Comment by gunan** [comment]
Merge PR, All CLAs were signed by authors.

**Comment by gunan** [comment]
Infra flake.
Jenkins, test this please.

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by gunan** [comment]
@yifeif the version update you made broke versions_test:
https://ci.tensorflow.org/job/tensorflow-pull-requests-cpu/2787/console

Could you take a look?
Maybe we need to change the regex that test checks?

File: tensorflow/core/public/version.h [hunk 1]
```diff
@@ -20,7 +20,7 @@ limitations under the License.
 
 #define TF_MAJOR_VERSION 0
 #define TF_MINOR_VERSION 12
-#define TF_PATCH_VERSION head
+#define TF_PATCH_VERSION 0-rc0
```
Comment (vrv) [inline tensorflow/core/public/version.h]: I wonder if this should be 'post' or something to indicate it's after 0.12 ?

Comment (yifeif) [inline tensorflow/core/public/version.h]: yea, we change the version test in this PR to check that we have a '-' before rc, which 'head' or 'post' will break. I'll need to update both.

Comment (gunan) [inline tensorflow/core/public/version.h]: Wait, I missed this.
I thought the agreement was "0.head"
0 as major version, "head" as patch version?

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by gunan** [comment]
One more to go.
@yifeif, could you also check tensorflow/core:util_semver_test ?

**Pull request merged**